spaceship/lib/spaceship/client.rb

```diff
@@ -28,19 +28,41 @@ class Client
     # /tmp/spaceship[time]_[pid].log by default
     attr_accessor :logger
 
+    # Base class for errors that want to present their message as
+    # preferred error info for fastlane error handling. See:
+    # fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb
+    class BasicPreferredInfoError < StandardError
+      def preferred_error_info
+        message ? [message] : nil
+      end
+    end
+
     # Invalid user credentials were provided
-    class InvalidUserCredentialsError < StandardError; end
+    class InvalidUserCredentialsError < BasicPreferredInfoError; end
 
     # Raised when no user credentials were passed at all
-    class NoUserCredentialsError < StandardError; end
+    class NoUserCredentialsError < BasicPreferredInfoError; end
 
-    class UnexpectedResponse < StandardError; end
+    class UnexpectedResponse < StandardError
+      attr_reader :error_info
+
+      def initialize(error_info = nil)
+        super(error_info)
+        @error_info = error_info
+      end
+
+      def preferred_error_info
+        return nil unless @error_info.kind_of?(Hash) && @error_info['resultString']
+
+        [@error_info['resultString'], @error_info['userString']].compact.uniq
```

Comment:
Hm. I originally had a comment here, but then felt like it was redundant information relative to the docs about what `compact` and `uniq` do.

e.g. "We `compact` because we want nil values removed, and we `uniq` to remove duplicate values."